### ISSUE TITLE
nil pointer dereference with kuttl 0.4.0

### ISSUE THREAD

**Issue by gpaul** [kudobuilder/kuttl#202]
I've just run into the following nil pointer crash and thought I'd report it:

```
    kuttl/harness/registry: logger.go:42: 17:25:15 | registry | Creating namespace: kudo-test-poetic-troll
    kuttl/harness/registry: logger.go:42: 17:25:15 | registry/0-create | starting test step 0-create
    kuttl/harness/registry: logger.go:42: 17:25:15 | registry/0-create | running command: "helm init --client-only"
    kuttl/harness/registry: logger.go:42: 17:25:15 | registry/0-create | running command: [helm init --client-only]
    kuttl/harness/registry: logger.go:42: 17:25:15 | registry/0-create | $HELM_HOME has been configured at /tekton/home/.helm.
    kuttl/harness/registry: logger.go:42: 17:25:15 | registry/0-create | Not installing Tiller due to 'client-only' flag having been set
    kuttl/harness/registry: logger.go:42: 17:25:15 | registry/0-create | running command: "helm delete --purge docker-registry"
    kuttl/harness/registry: logger.go:42: 17:25:15 | registry/0-create | running command: [helm delete --purge docker-registry]
    kuttl/harness/registry: logger.go:42: 17:25:26 | registry/0-create | Error: release: "docker-registry" not found
...
    kuttl/harness/registry: logger.go:42: 17:26:52 | registry/0-create | Error: release docker-registry failed: Get https://10.96.0.1:443/api/v1/namespaces/kudo-test-poetic-troll: dial tcp 10.96.0.1:443: connect: connection refused
    kuttl/harness/registry: case.go:156: exit status 1
    kuttl/harness/registry: case.go:156: Get "https://127.0.0.1:32768/api?timeout=32s": EOF
    kuttl/harness/registry: case.go:159: failed in step 0-create
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x131e945]

goroutine 8260 [running]:
github.com/kudobuilder/kuttl/pkg/test/utils.(*RetryClient).List.func1(0x1936320, 0xc000046048, 0x0, 0x0)
	/Users/kensipe/projects/go/src/github.com/kudobuilder/kuttl/pkg/test/utils/kubernetes.go:228 +0x25
github.com/kudobuilder/kuttl/pkg/test/utils.Retry.func1(0xc0004670c0, 0x1936320, 0xc000046048, 0xc0006d5b60, 0xc0006d5ce0)
	/Users/kensipe/projects/go/src/github.com/kudobuilder/kuttl/pkg/test/utils/kubernetes.go:110 +0x3e
created by github.com/kudobuilder/kuttl/pkg/test/utils.Retry
	/Users/kensipe/projects/go/src/github.com/kudobuilder/kuttl/pkg/test/utils/kubernetes.go:109 +0xf7
Collecting kind cluster logs.
```

I believe this was with kuttl version 0.4.0.

**Comment by kensipe** [comment]
hmm... this is clearly a nil being passed in as a `func`.  Test confirm as well.
It is easy to defend against this... but it is hard to track down the function that passed a nil.  It is clearly a `List`  which means `r.Client.List`.   This is in a go routine... it is easily possible that the client was torn down on the main thread prior to this thread running.   

tough problem!  thanks for logging this!   And let this be a lesson around passing a reference to go routine which stays in a forever loop :)    
All code review was on v0.4.0.   I'll add a fix here for a v0.4.1, v0.5.2, then add for v0.6.1 and master.

**Comment by kensipe** [comment]
not as easy as I hoped...  the issue is the `fn` is an anonymous function which is of course not nil... it has calls internal which when called are nil.   We will have a solution soonish with tests.
